pkg/security/probe/probe_ebpf.go

```diff
@@ -646,10 +640,26 @@ func (p *EBPFProbe) handleEvent(CPU int, data []byte) {
 		p.playSnapshot(state == playSnapShotState)
 	}
 
-	offset := 0
-	event := p.zeroEvent()
+	var (
+		offset        = 0
+		event         = p.zeroEvent()
+		dataLen       = uint64(len(data))
+		relatedEvents []*model.Event
+		newEntryCb    = func(entry *model.ProcessCacheEntry, err error) {
+			relatedEvent := newEBPFEventFromPCE(entry, p.fieldHandlers)
+
+			if err != nil {
+				var errResolution *path.ErrPathResolution
+				if errors.As(err, &errResolution) {
+					relatedEvent.SetPathResolutionError(&relatedEvent.ProcessCacheEntry.FileEvent, err)
+				} else {
+					return
+				}
+			}
 
-	dataLen := uint64(len(data))
+			relatedEvents = append(relatedEvents, relatedEvent)
+		}
+	)
```

Comment:
Instead of defining this cb here and pass it on all subsequent resolver calls, would it be possible to register it at init time and keep it in the resolver itself, maybe as well as the "pending events" that have to be dispatched (if any) later on? Maybe it's not possible, but if it is I'll personally prefer to avoid passing the same cb (or nil) to all the func calls that you've updated

Reply:
As discuss offline it will be more difficult to handle case where we want to handle the fallback entry and when we don't with a "register listener" approach.